superset/views/core.py

```diff
@@ -2075,7 +2075,9 @@ def select_star(self, database_id, table_name, schema=None):
         schema = utils.parse_js_uri_path_item(schema, eval_undefined=True)
         table_name = utils.parse_js_uri_path_item(table_name)
         # Check that the user can access the datasource
-        if not self.appbuilder.sm.can_access_datasource(database, table_name, schema):
+        if not self.appbuilder.sm.can_access_datasource(
+            database, Table(table_name, schema), schema
```

Comment:
The `schema` is optional in the `Table(...)` since we're also providing a fallback. Regardless I think we should be explicit as possible  when defining a table and include the schema if defined.